crates/runtime/src/traphandlers.rs

```diff
@@ -603,13 +735,90 @@ mod tls {
     }
 }
 
+/// A module for handling faults on Fuchsia
+#[cfg(target_os = "fuchsia")]
+mod fuchsia {
+    use super::*;
+
+    // Be very careful here. There is a contract with the stack in setup_fault_handler
+    pub extern "C" fn fault_handler(
+        regs: *const sys::zx_thread_state_general_regs_t,
+        report: *const sys::zx_exception_report_t,
+    ) {
+        let handled = tls::with(|info| {
```

Comment:
Should this perhaps have a TODO to do something with `handled`? If it's `false` this may infinitely try to handle the fault here?